pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/admin/UserAdminServiceTest.java

```diff
@@ -35,122 +28,48 @@
     }
 
     @After
     public void cleanup() {
-        for (int groupRoleId: createdGroupRoles) {
-            UserAdminService.deleteGroupRole(groupRoleId);
-        }
-    }
-
-    private int createTestUser(String email, int roleId) {
-        int userId = createUser(email);
-        List<Integer> roleIds = new ArrayList<>();
-        if (roleId != -1) {
-            roleIds.add(roleId);
-        }
-        createdUserRoles.put(userId, roleIds);
-        return userId;
+        testUsers.deleteStudyAdminAndRoles();
     }
 
-    private int createUser(String email) {
-        String name = email.split("@")[0];
-        UserDto userDto = new UserDto();
-        userDto.setName(name);
-        userDto.setEmail(email);
-        userDto.setIsActive(1);
-        UserDao userDao = new UserDao();
-        return userDao.create(userDto);
-    }
-
-    private int addUserRole(int userId, int roleId, int groupId) {
-        int userRoleId = UserAdminService.addUserRole(userId, roleId, groupId);
-        addRoleForUser(roleId, userId);
-        return userRoleId;
-    }
-
-    private void addRoleForUser(int roleId, int userId) {
-        List<Integer> roleIds = createdUserRoles.get(userId);
-        List<Integer> newRoleIds = new ArrayList<>(roleIds);
-        newRoleIds.add(roleId);
-        createdUserRoles.put(userId, newRoleIds);
-    }
-
-    private void removeRoleForUser(int roleId, int userId) {
-        List<Integer> roleIds = createdUserRoles.get(userId);
-        List<Integer> newRoleIds = new ArrayList<>(roleIds);
-        newRoleIds.removeAll(List.of(roleId));
-        createdUserRoles.put(userId, newRoleIds);
-    }
-
-    private int addGroupRole(int roleId, int adminRoleId) {
-        int groupRoleId = UserAdminService.addGroupRole(studyGroupId, roleId, adminRoleId);
-        createdGroupRoles.add(groupRoleId);
-        return groupRoleId;
-    }
-
-    private void removeGroupRole(int groupRoleId) {
-        UserAdminService.deleteGroupRole(groupRoleId);
-        createdGroupRoles.remove(groupRoleId);
-    }
 
     public void testGetRoleId() {
         // temporary to understand which roles are handled by liquibase
-        log.info("TEMP: all roles: {}", String.join("\n", getAllRoles()));
+        log.info("TEMP: all roles: {}", String.join("\n", UserAdminTestUtil.getAllRoles().keySet()));
         int roleId = UserAdminService.getRoleId("upload_onc_history");
         Assert.assertTrue(roleId > 0);
     }
 
     @Test
     public void testVerifyOperatorGroup() {
-        int roleId = UserAdminService.getRoleId("dashboard_view");
-        Assert.assertTrue(roleId > 0);
-        int userId = createTestUser("test_admin1@study.org", roleId);
+        String roleName = "dashboard_view";
+        int userId = -1;
+        try {
+            userId = testUsers.createTestUser("test_admin1@study.org", List.of(roleName));
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail();
```

Comment:
Maybe add some text to the `fail()` so that whoever's looking at the test result knows a bit more about what was expected?

Reply:
Will do. FWIW, this (and similar in this file) are catching unlikely internal errors, not testing something per se, thus the printStackTrace. It is nicer to have the stack trace in the output than to try to deduce the error from the Surefire logs.